packages/yew-agent-macro/src/lib.rs

```diff
@@ -0,0 +1,30 @@
+use proc_macro::TokenStream;
+use syn::parse_macro_input;
+
+mod agent_fn;
+mod oneshot;
+mod reactor;
+
+use agent_fn::{AgentFn, AgentName};
+use oneshot::{oneshot_impl, OneshotFn};
+use reactor::{reactor_impl, ReactorFn};
+
+#[proc_macro_attribute]
+pub fn reactor(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let item = parse_macro_input!(item as AgentFn<ReactorFn>);
+    let attr = parse_macro_input!(attr as AgentName);
+
+    reactor_impl(attr, item)
+        .unwrap_or_else(|err| err.to_compile_error())
+        .into()
+}
+
+#[proc_macro_attribute]
+pub fn oneshot(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let item = parse_macro_input!(item as AgentFn<OneshotFn>);
+    let attr = parse_macro_input!(attr as AgentName);
+
+    oneshot_impl(attr, item)
+        .unwrap_or_else(|err| err.to_compile_error())
+        .into()
+}
```

Comment:
Why can't the gloo-worker macros be used here?

Reply:
Because `gloo_worker_macros` references `gloo_worker` in generated code. `yew-agent-macro` needs to reference `yew-agent` in generated code.

In addition, it is not possible to expose anything other than procedural macros from a macro crate.
A new crate is needed from the gloo side to expose the implementations of procedural macros.

The macro implementation of yew-agent and gloo-worker also differs in diagnostics (agents are named agents instead of workers).

Reply:
I would prefer to expose `gloo-worker-macro-impl` crate (that can also be compiled to wasm) and then call it from there. It's fine for now. I presume we'd need this crate this yew-agent-macro either way so I'm fine with merging duplicate implementation of it.